src/chart/model/scale.model.ts

```diff
@@ -67,6 +67,11 @@ export class ScaleModel extends ViewportModel {
 
 	xConstraints: Constraints[] = [];
 
+	maxZoomReached: {
+		zoomIn: boolean;
+		zoomOut: boolean;
+	} = { zoomIn: false, zoomOut: false };
```

Comment:
Additionally, default values are false. Are you sure that it is correct? What if client opened app with saved max scale?

Reply:
`zoomConstraints are needed to limit the max and min amount of candles if we have them (10 for zoom in, and the amount of candles which will fit with 1px width for zoom out)` 
I'd suggest a solution, that will calculate max and min amount of candles to zoomX value, and it will be source of truth for amount of candles and value maxZoomReached that you are trying to implement

Reply:
> > Looks like max zoom value for now is controlled by `scalePostProcessor` and depends on `xContraints` You should use only one way to control it. If you want to control it via `scaleModel.maxZoomReached` object, you should use it in whole app
> 
> Not sure I understand, `xConstraints` are needed to not go outside of candles (we should have at least 2 candles in viewport)
> 
> `zoomConstraints` are needed to limit the max and min amount of candles if we have them (10 for zoom in, and the amount of candles which will fit with 1px width for zoom out)

I'd suggest a solution, that will calculate max and min amount of candles to zoomX value, and it will be source of truth for amount of candles and value maxZoomReached that you are trying to implement

Reply:
> > Additionally, default values are false. Are you sure that it is correct? What if client opened app with saved max scale?
> 
> It works the way that if, for example, zoom in is maxed, then we can't do zoom in, but zoom out is still available, this is why we check not only the state of maxZoom but the action we do:
> 
> <img alt="Code_hrNWBJf7q1" width="538" src="https://private-user-images.githubusercontent.com/23086580/318417707-89e2f9ff-93cd-4e99-8dc1-477a33304f5e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5ODE1ODIsIm5iZiI6MTcxMTk4MTI4MiwicGF0aCI6Ii8yMzA4NjU4MC8zMTg0MTc3MDctODllMmY5ZmYtOTNjZC00ZTk5LThkYzEtNDc3YTMzMzA0ZjVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDE0MjEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNzJjYThlMGZhYWNlNWQwMWM2ZjA1Y2UzYTY2MmE3ZjA3MzViZGJhNmFlMTk4YWIyNzkzZmY1MmU2Zjg1NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NTKbNoYkSx-pDzvsqll46turx6B_P-pE1k4ZRlOpfqk">

means we have to rename this object because it doesn't reflect the functionality `zoomReached: { max: false, min: false }`, and I'd like to see it in scale model state. Also, my point here is that this value should be set in contractor, based on loaded app's zoomX